arches/app/views/api/__init__.py

```diff
@@ -2383,3 +2383,128 @@ def delete(self, request, identifier=None):
+    def post(self, request, identifier=None):
+        if identifier:
+            return JSONErrorResponse(
+                _("User Preference creation failed"),
+                _(
+                    "POST request should not have a userpreferenceid provided in the URL"
+                ),
+                status=400,
+            )
+
+        try:
+            user_pref_json = JSONDeserializer().deserialize(request.body)
+        except ValueError:
+            return JSONErrorResponse(
+                _("Invalid JSON data"),
+                _("The User Preference API was sent invalid JSON"),
+                status=400,
+            )
+
+        if user_pref_json:
+            if user_pref_json["userpreferenceid"]:
+                JSONErrorResponse(
+                    _("Incorrect User Preference json data"),
+                    _(
+                        "POST REST request should not have a userpreferenceid provided in the JSON data."
+                    ),
+                    status=400,
+                )
+            else:
+                try:
+                    models.User.objects.get(username=user_pref_json["user"])
+                except:
+                    return JSONErrorResponse(
+                        _("Invalid user"),
+                        _("The User Preference API includes an invalid user."),
+                        status=400,
+                    )
+                new_user_preference = models.UserPreference()
+                new_user_preference.userpreferenceid = uuid.uuid4()
+                new_user_preference.user = models.User.objects.get(
+                    username=user_pref_json["user"]
+                )
+                new_user_preference.preferencename = user_pref_json["preferencename"]
+                new_user_preference.config = user_pref_json["config"]
+                new_user_preference.save()
+                return JSONResponse(new_user_preference, status=201)
+
+        return JSONErrorResponse(_("No json request payload"), status=400)
+
+    @method_decorator(group_required("Application Administrator", raise_exception=True))
+    def delete(self, request, identifier=None):
+        if identifier:
+            user_preference = None
+            try:
+                user_preference = models.UserPreference.objects.get(pk=identifier)
+            except ObjectDoesNotExist:
+                return JSONErrorResponse(
+                    _("User Preference delete failed"),
+                    _("User Preference does not exist"),
+                    status=404,
+                )
+
+            try:
+                user_preference.delete()
+            except Exception as e:
+                logger.error(e)
+                return JSONErrorResponse(
+                    _("User Preference delete failed"),
+                    _("An error occurred when trying to delete the spatialview"),
+                    status=500,
+                )
+
+        else:
```

Comment:
Same as above, no need to handle for this if this is just a method on UserPreferenceDetailView with a mandatory identifier component